app_base.htm

```diff
@@ -0,0 +1,82 @@
+          </div>
+          <ul class="nav navbar-nav">
+            <li class="${ 'active' if request.dmp.page == 'index' and request.dmp.app == 'homepage' else '' }"><a href="/index/">Home</a></li>
+            <li class="${ 'active' if request.dmp.page == 'sections' else '' }"><a href="/sections/">Sections</a></li>
+            <li class="${ 'active' if request.dmp.page == 'about' else '' }"><a href="/about/">About</a></li>
+            <li class="${ 'active' if request.dmp.page == 'contact' else '' }"><a href="/contact/">Contact</a></li>
+            <li class="${ 'active' if request.dmp.page == 'terms' else '' }"><a href="/terms/">Terms</a></li>
+            <li class="${ 'active' if request.dmp.page == 'faq' else '' }"><a href="/faq/">FAQ</a></li>
+            <li class="${ 'active' if request.dmp.app == 'catalog' else '' }"><a href="/catalog/">Products</a></li>
+          </ul>
+          <ul class="nav navbar-nav navbar-right">
+            % if request.user.is_authenticated:
+              <li class="dropdown"><a class="dropdown-toggle" data-toggle="dropdown" href="#">${ request.user }<span class="caret"></span></a>
+                <ul class="dropdown-menu">
+                    <li><a href="/account/index/">My Account</a></li>
+                    <li><a href="/account/logout/">Log Out</a></li>
+                </ul>
+              </li>
+            % else:
+              <li class="${ 'active' if request.dmp.page == 'signup' else '' }"><a href="/account/signup/"><span class="glyphicon glyphicon-user"></span> Sign Up</a></li>
+              <li class="${ 'active' if request.dmp.page == 'login' else '' }"><a href="/account/login/"><span class="glyphicon glyphicon-log-in"></span> Login</a></li>
+            % endif
+          </ul>
+        </div>
+    </nav>
+</%block>
+
+<%block name="content_left">
+    <div id='category_list'>
+        <%
+            category_list = cmod.Category.objects.all()
+            product_list = cmod.Product.objects.all()
+        %>
+        <ul>
+            % if cat_id == 0:
+                <li id='all_products'><a class='active' href='/catalog/index/'> All Products </a></li>
+            % else:
+                <li id='all_products'><a href='/catalog/index/'> All Products </a></li>
+            % endif
+            % for c in category_list:
+                % if cat_name == c.name:
+                    <li><a class='active' href='/catalog/index/${ c.id }'/> ${ c.name } </a></li>
+                % else:
+                    <li><a href='/catalog/index/${ c.id }'/> ${ c.name } </a></li>
+                % endif
+            % endfor
+        </ul>
+    </div>
+</%block>
+
+<%block name="content_right">
+    <div id='right_column'>
+        <h3>Recently Viewed</h3>
+        <ol>
+            <%
+                if request.dmp.page == 'detail':
+                    printlist = request.last_five[1:6]
```

Comment:
Possibly try to avoid hard coding numbers in the code. This would help in case in the future you'd want to change the amount shown.. for example, if you wanted to change from 5 products to 3 products it'd be a quick fix. Right now it's only in one area, but for future reference.

Reply:
I feel that you are being very nitpicky right now

Reply:
I agree with @MatthewDembinski21 on this one. Creating a variable to use instead of hard coding the numbers could be useful for expanding later.

Reply:
Thanks for the input guys